### ISSUE TITLE
STAR Selection: Result predictable order of columns 

### ISSUE THREAD

**Issue by pgollangi** [pgollangi/FireQL#2]
When all fields of Firestore documents selected using STAR selection (`select * from users`), the order of columns resulted is not always same. 
This is because when STAR selected, `FireQL` don't explicitly request column selections to get all fields of Firestore collection and uses data returned by [Firestore.DocumentSnapshot](https://pkg.go.dev/cloud.google.com/go/firestore#DocumentSnapshot.Data) which is a map field and value pair where order of key is not always same.

One way to return the same columns always would be by sorting columns (map keys) returned by `DocumentSnapshot.Data`

**Comment by rafiramadhana** [comment]
does it mean we need to sort the columns (`selectedColumns`) before exiting this if-else statement [select.go#L109](https://github.com/pgollangi/FireQL/blob/main/pkg/select/select.go#L109)?

```
if starIdx != -1 {
    ...
        }
        starIdx++
    }
    // TODO: Sort selectedColumns before exiting this if-else statement
}
```

**Comment by pgollangi** [comment]
Hi @rafiramadhana thanks a lot for your interest in this project. 

Sorting `selectedColumns` would possibly change the position of other selected columns along with `*`. For example `select user_id, *, account_id from users` should list user_id first and account_id last in resulted columns and remaining all columns in order in place of `*`.

You could possible collect all fields from data and sort then add it to `selectedColumns` rather than adding it on-the-fly.

**Comment by rafiramadhana** [comment]
@pgollangi i see, to spawn a _mocked firestore_ in local, is it enough by doing these steps?
- Run [gcloud beta emulators](https://cloud.google.com/sdk/gcloud/reference/beta/emulators/firestore/start) (ref: [select/select_test.go](https://github.com/pgollangi/FireQL/blob/main/pkg/select/select_test.go#L123))
- Populate the collection using [data/users.json](https://github.com/pgollangi/FireQL/blob/main/test/data/users.json)

btw is there any UI/CLI app to interact with the _mocked firestore_? like for seeing data, updating data, adding collection, etc

**Comment by pgollangi** [comment]
@rafiramadhana yes thats right. I generally follow [programatical way](https://cloud.google.com/firestore/docs/manage-data/add-data) of manage firestore data, I found it easy. however there are many open-source and commercial tools to connect firestore.
